libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -745,36 +745,54 @@ bool ContractCompiler::visit(InlineAssembly const& _inlineAssembly)
 				else if (m_context.isStateVariable(decl))
 				{
 					auto const& location = m_context.storageLocationOfVariable(*decl);
-					if (ref->second.isSlot)
+					if (ref->second.suffix == "slot")
```

Comment:
Do you think it was easier to do it as a inline method `isSlot()` that would do actual check `suffix == "slot"`?

I am fine with leaving as is in this PR. Can be done as a separate refactor if you think it is worth doing it.

Reply:
Actually it is not used that much, so probably not even worth it

Reply:
In the end, these suffixes are exactly the same as the ones in `makeStackItems` in Types.cpp. So maybe at some point, we can just check whether or not they are available and thus support all of them.